src/L.Control.Elevation.js

```diff
@@ -669,6 +669,7 @@ L.Control.Elevation = L.Control.extend({
         }
         if (layer) {
             layer.on("mousemove", this._handleLayerMouseOver.bind(this));
+            layer.on("mouseout", this._hidePositionMarker.bind(this));
```

Comment:
There is a problem if the elevation profile is removed from the Leaflet map. The layer would still have listeners pointing to an elevation profile that no longer exists. I think we should take care of removing listeners too.

Try this. Here, change lines 670-673 to:
```javascript
        if (layer) {
            this._layer = layer;
            layer.on("mousemove", this._handleLayerMouseOver, this).
            on("mouseout", this._handleLayerMouseOut, this);
        }
```
And add in onRemove at top of file:

```javascript
        if (this._layer) {
            this._layer.off("mousemove", this._handleLayerMouseOver, this).
            off("mouseout", this._handleLayerMouseOut, this);
        }
```

Reply:
Good point. I guess in theory `addData` could be called multiple times, so we would need to keep a list of layers. `clear` should probably also remove listeners.

I won't update this right now, maybe later. Feel free to share an alternative PR and I'll close mine.